### ISSUE TITLE
Minecraft Bedrock/Pocket Edition?!

### ISSUE THREAD

**Issue by iSilviu-debug** [haveachin/infrared#44]
How difficult would it be to support Geyser? (Minecraft Bedrock / Pocket Edition)
**Link**: https://geysermc.org/

Thank you! 🌵

**Comment by iSilviu-debug** [comment]
Protocol tip: GeyserMC uses **UDP** and not **TCP**

**Comment by haveachin** [comment]
It should work, if geyser translates to normal java edition protocol.
Your setup should be: [mc clients] -> [geyser proxy] -> [infrared] -> [mc server]

**Comment by tdeverx** [comment]
Seems I discovered Geyser at just the right time! My first thought was infrared, how have you approached the implementation? Are we talking something as simple as adding 2 separate hosts / ports in the config?

java.exaple.com : 25565
bedrock.example.com : 19132

**Comment by haveachin** [comment]
The core logic is already implemented in the rewrite branch. Infrared only checks the handshake packet and routes the traffic according to the server address. It does not interfere with client-server authentication nor does it listen or decrypt any communication between client and server after the initial handshake. Here is the core implementation for Bedrock https://github.com/haveachin/infrared/tree/rewrite/internal/pkg/bedrock

**Comment by haveachin** [comment]
You could already test it if you compile the code yourself. There is also a Makefile and a Dockerfile/docker-compose.yml if you know how to use those. But I would only recommend using it for testing purposes.

**Comment by tdeverx** [comment]
Yeah I noticed! But unfortunately, I'm running my containers on Synology and I don't know the first thing about compiling code myself, and that's _excluding_ docker-compose 😂 (I'm relatively new to docker hahaha)
